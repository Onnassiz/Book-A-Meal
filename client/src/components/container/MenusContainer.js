import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { getMealsInDailyMenu } from '../../reducer/menus/actionsDispatcher';
import { deleteFromCart, addToCart, updateCart, emptyCart } from '../../reducer/cart/actionsDispatcher';

import Menus from '../presentation/Menus';

export const mapStateToProps = state => ({
  menus: state.menus,
  cart: state.cart,
  orders: state.orders,
  user: state.user,
  formState: state.formState,
});

export function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    addToCart,
    updateCart,
    emptyCart,
    deleteFromCart,
    getMealsInDailyMenu,
  }, dispatch);
}

export default withRouter(connect(
  mapStateToProps,
  mapDispatchToProps,
)(Menus));
