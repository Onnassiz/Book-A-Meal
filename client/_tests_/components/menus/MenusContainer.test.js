import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import MenusContainer, { mapStateToProps, mapDispatchToProps } from '../../../src/components/container/MenusContainer';

const mockStore = configureStore();

const store = mockStore({
  user: {},
  formState: { isLoading: false },
  menus: {},
  cart: {},
  orders: {},
});

const initialState = {
  user: {},
  formState: { isLoading: false },
};

const wrapper = shallow(<MenusContainer store={store} />);

describe('<MenusContainer />', () => {
  it('Should render the MenusContainer container successfully', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).addToCart();
    expect(mapStateToProps(initialState).user).toEqual({});
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
