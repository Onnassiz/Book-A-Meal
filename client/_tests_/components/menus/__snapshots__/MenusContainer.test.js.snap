// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenusContainer /> Should render the MenusContainer container successfully 1`] = `
<Route
  render={[Function]}
/>
`;
